### PR TITLE
[Communication] Change SMS APIs to be pageable

### PR DESCRIPTION
**Pull request by ankitarorabit**
* As per the APIView feedback, APIs returning collections should be pageable even if the APIs are not

### CODE REVIEW

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java [hunk 1]
```diff
@@ -81,11 +79,9 @@ public void sendSmsToGroupWithOptions(HttpClient httpClient) {
...
 
         // Action & Assert
-        StepVerifier.create(asyncClient.sendWithResponse(FROM_PHONE_NUMBER, Arrays.asList(TO_PHONE_NUMBER, TO_PHONE_NUMBER), MESSAGE, options))
-            .assertNext((Response<Iterable<SmsSendResult>> response) -> {
-                for (SmsSendResult result : response.getValue()) {
-                    assertHappyPath(result);
-                }
+        StepVerifier.create(asyncClient.send(FROM_PHONE_NUMBER, Arrays.asList(TO_PHONE_NUMBER, TO_PHONE_NUMBER), MESSAGE, options).next())
+            .assertNext(result -> {
+                assertHappyPath(result);
```
Comment (minnieliu) [inline sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:84]: Wondering why the for-loop was removed here?

Comment (ankitarorabit) [reply sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:84]: The APIs result type now is PagedFlux<SmsSendResult> (not the Response<iterable<SmsSendResult>> which can be looped).. and I believe the introduction of .next() in the create step takes care of looping through the objects .. (I mostly followed the other tests in PhoneNumber for flux APIs :))

**Review by minnieliu — APPROVED** [review]

File: sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:84]: Hmm I see, I think next only gets the next value in the page and does not loop through all the objects in the list

Comment (minnieliu) [reply sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:84]: I think you can do something like this: 

`
 PagedIterable<SmsSendResult> sendResults = new PagedIterable<>(result);
                sendResults .iterableByPage().forEach(resp -> {
                       assertHappyPath(result);
                });
`

Comment (ankitarorabit) [reply sdk/communication/azure-communication-sms/src/test/java/com/azure/communication/sms/SmsAsyncClientTests.java:84]: Interesting, I thought I saw that the validation actually checked for 2 results but maybe I was mistaken, will use this code instead.. thanks!

**Pull request merged**